WalletWasabi/Blockchain/TransactionOutputs/CoinsRegistry.cs

```diff
@@ -293,4 +242,38 @@ public ICoinsView AsAllCoinsView()
 	public ICoinsView Unspent() => AsCoinsView().Unspent();
 
 	IEnumerator IEnumerable.GetEnumerator() => AsCoinsView().GetEnumerator();
+
+	public bool Seen(uint256 txid)
+	{
+		lock (Lock)
+		{
+			return Transactions.ContainsKey(txid);
+		}
+	}
+
+	public SmartCoin[] GetMyInputs(SmartTransaction transaction)
+	{
+		var inputs = transaction.Transaction.Inputs.Select(x => x.PrevOut).ToArray();
+
+		var myInputs = new List<SmartCoin>();
+		lock (Lock)
+		{
+			foreach (var input in inputs)
+			{
+				if (!Transactions.TryGetValue(input.Hash, out var transactionInternal))
+				{
+					continue;
+				}
+
+				if(transactionInternal.Coins.TryGetValue(input, out var coin))
+				{
+					myInputs.Add(coin);
+				}
+			}
+		}
+
+		return myInputs.ToArray();
+	}
```

Comment:
These 2 functions, and their usage in the `TransactionProcessor` are the changes from #11513

Reply:
Merged into master